enginetest/enginetests.go

```diff
@@ -47,129 +48,126 @@ import (
-		for _, tt := range KeylessQueries {
-			TestQuery(t, harness, engine, tt.Query, tt.Expected, tt.ExpectedColumns)
+		for _, tt := range queries.KeylessQueries {
+			TestQuery(t, harness, tt.Query, tt.Expected, tt.ExpectedColumns, nil)
 		}
 	}
 }
 
 // Tests a variety of geometry queries against databases and tables provided by the given harness.
 func TestSpatialQueries(t *testing.T, harness Harness) {
-	engine := NewSpatialEngine(t, harness)
-	defer engine.Close()
-
-	for _, tt := range SpatialQueryTests {
-		TestQuery(t, harness, engine, tt.Query, tt.Expected, tt.ExpectedColumns)
+	harness.SetSetup(spatialSetup...)
+	e := mustNewEngine(t, harness)
+	for _, tt := range queries.SpatialQueryTests {
+		TestQueryWithEngine(t, harness, e, tt)
 	}
 }
 
 // Tests a variety of geometry queries against databases and tables provided by the given harness.
 func TestSpatialQueriesPrepared(t *testing.T, harness Harness) {
-	engine := NewSpatialEngine(t, harness)
-	defer engine.Close()
-	for _, tt := range SpatialQueryTests {
-		TestPreparedQuery(t, harness, engine, tt.Query, tt.Expected, tt.ExpectedColumns)
+	harness.SetSetup(spatialSetup...)
+	e := mustNewEngine(t, harness)
+	for _, tt := range queries.SpatialQueryTests {
+		TestPreparedQueryWithEngine(t, harness, e, tt)
 	}
 
-	newEngine := func() *sqle.Engine {
-		return NewSpatialEngine(t, harness)
-	}
-	for _, tt := range SpatialDeleteTests {
-		runWriteQueryTest(t, harness, tt, newEngine)
+	for _, tt := range queries.SpatialDeleteTests {
+		runWriteQueryTestPrepared(t, harness, tt)
 	}
-	for _, tt := range SpatialInsertQueries {
-		runWriteQueryTest(t, harness, tt, newEngine)
+	for _, tt := range queries.SpatialInsertQueries {
+		runWriteQueryTestPrepared(t, harness, tt)
 	}
-	for _, tt := range SpatialUpdateTests {
-		runWriteQueryTest(t, harness, tt, newEngine)
+	for _, tt := range queries.SpatialUpdateTests {
+		runWriteQueryTestPrepared(t, harness, tt)
 	}
-
-	t.Run("create table with NULL default values for geometry types", func(t *testing.T) {
-		ctx := NewContext(harness)
-
-		TestQuery(t, harness, engine, "CREATE TABLE null_default (pk int NOT NULL PRIMARY KEY, v1 geometry DEFAULT NULL, v2 linestring DEFAULT NULL, v3 point DEFAULT NULL, v4 polygon DEFAULT NULL)",
-			[]sql.Row{{sql.NewOkResult(0)}}, nil)
-		db, err := engine.Analyzer.Catalog.Database(ctx, "mydb")
-		require.NoError(t, err)
-
-		_, ok, err := db.GetTableInsensitive(ctx, "null_default")
-		require.NoError(t, err)
-		require.True(t, ok)
-	})
-
 }
 
 // Tests join queries against a provided harness.
 func TestJoinQueries(t *testing.T, harness Harness) {
-	engine := NewEngine(t, harness)
-	CreateIndexes(t, harness, engine)
-	createForeignKeys(t, harness, engine)
-
-	for _, tt := range JoinQueryTests {
-		TestQuery(t, harness, engine, tt.Query, tt.Expected, tt.ExpectedColumns)
+	harness.SetSetup(MydbData, MytableData, Pk_tablesData, OthertableData)
+	for _, tt := range queries.JoinQueryTests {
```

Comment:
What would be even better is if these query bundles declared their setup dependencies themselves. Can do that in a later pass though